tests/test_converter.py

```diff
@@ -789,6 +829,98 @@ def test_barrier_not_supported(self, recorder):
             " PennyLane, and has not been added to the template."
         )
 
+    @pytest.mark.parametrize("invalid_param", ["wires", "params"])
+    def test_params_and_wires_not_valid_param_names(self, invalid_param):
+        """Test that ambiguous parameter names 'wires' and 'params' in the Qiskit
+        QuantumCircuit raise an error"""
+
+        parameter = Parameter(invalid_param)
+
+        qc = QuantumCircuit(2, 2)
+        qc.rx(parameter, 0)
+        qc.rx(0.3, 1)
+        qc.measure_all()
+
+        with pytest.raises(RuntimeError, match="this argument is reserved"):
+            load(qc)(0.3)
+
+    def test_kwarg_does_not_match_params(self):
+        """Test that if a parameter kwarg doesn't match any of the Parameter
+        names in a QuantumCircuit, an error is raised"""
+
+        parameter = Parameter("name1")
+
+        qc = QuantumCircuit(2, 2)
+        qc.rx(parameter, 0)
+        qc.rx(0.3, 1)
+        qc.measure_all()
+
+        # works with correct name
+        load(qc)(name1=0.3)
+
+        # raises error with incorrect name
+        with pytest.raises(TypeError, match="Got unexpected parameter keyword argument 'name2'"):
+            load(qc)(name2=0.3)
+
+    def test_too_many_args(self):
+        """Test that if too many positional arguments are passed to define Parameter values,
+        a clear error is raised"""
+
+        a = Parameter("a")
+        b = Parameter("b")
+
+        qc = QuantumCircuit(2, 2)
+        qc.rx(a, 0)
+        qc.rx(b, 1)
+        qc.measure_all()
+
+        with pytest.raises(TypeError, match="Expected 2 positional arguments but 3 were given"):
+            load(qc)(0.2, 0.4, 0.5)
+
+        with pytest.raises(TypeError, match="Expected 1 positional argument but 2 were given"):
+            load(qc)(0.4, 0.5, a=0.2)
+
+    def test_missing_argument(self):
+        """Test that if calling with missing arguments, a clear error is raised"""
+
+        a = Parameter("a")
+        b = Parameter("b")
+        c = Parameter("c")
+
+        qc = QuantumCircuit(2, 2)
+        qc.rx(a, 0)
+        qc.rx(b * c, 1)
+        qc.measure_all()
+
+        with pytest.raises(
+            TypeError, match="Missing 1 required argument to define Parameter values"
+        ):
+            load(qc)(0.2, 0.3)
+
+        with pytest.raises(
+            TypeError, match="Missing 1 required argument to define Parameter values"
+        ):
+            load(qc)(0.2, c=0.4)
+
+        with pytest.raises(
+            TypeError, match="Missing 2 required arguments to define Parameter values"
+        ):
+            load(qc)(b=0.3)
```

Comment:
note: should also add the old UI with too few arguments to this test (and the other similar ones)